compiler-core/src/type_/error.rs

```diff
@@ -655,6 +655,14 @@ pub enum Error {
         package: EcoString,
         location: SrcSpan,
     },
+
+    /// This happens when a type has no type arguments (for example `Int`) but
+    /// it is written with an explicit empty argument list: `Int()`.
```

Comment:
No such thing as an argument list.

I think it'd be more accurate to say that this error is for a type being used as a type constructor?